app/cdash/include/ctestparser.php

```diff
@@ -168,10 +169,30 @@ function ctest_parse($filehandle, $projectid, $expected_md5 = '')
     }
 
     // Figure out what type of XML file this is.
-    $xml_info = SubmissionUtils::get_xml_type($filehandle);
+    try {
+        $xml_info = SubmissionUtils::get_xml_type($filehandle, $filename);
```

Comment:
Given that the path passed to `validate_xml()` had to be modified to `storage_path("app/".$filename)`, should this path being passed to `get_xml_type()` be changed as well? If so, I think these two are the only usages of the `$filename` function argument, so better yet, modify the value being passed to `ctest_parse()` from `app/Jobs/ProcessSubmission.php`.

Reply:
@sbelsk, the `$xml_file` variable within the `get_xml_type` only seems to be used in the error conditions at the end. if no file type and schema file are found.  The content reading is all done using the `$filehandle` object.  I'm not sure if `$filehandle`  will ever *not* be valid yet.

Reply:
Makes sense. Might still be worth changing it for the logging in lines 191-193 `ctest_parse()` in case the XML validation fails.